docusaurus/docs/common-use-cases/providing-viewmodels.md

```diff
@@ -211,9 +211,53 @@ class ViewModelFactory(
 }
 ```
 
-As seen above, if you want a single function for all viewModel types, then you need a single `ViewModelFactory` that can create all your types of ViewModels. Alternatively, you could have one function for each ViewModel (example: `profileViewModel()` to get `ProfileViewModel` by using a `ProfileViewModelFactory`).
+As seen above, if you want a single function for all viewModel types, then you need a single `ViewModelFactory` that can create all your types of ViewModels. This does mean that every time you add a new ViewModel to your code base, you'll have to come here and add that entry to the `when` statement above. Nothing is perfect, and this is definitely one reason to make you want to use a Dependency Injection framework like Hilt, so that this is handled for you.
 
-This does mean that every time you add a new ViewModel to your code base, you'll have to come here and add that entry to the `when` statement above. Nothing is perfect, and this is definitely one reason to make you want to use a Dependency Injection framework like Hilt, so that this is handled for you.
+Alternatively, you could have one function for each ViewModel (example: `profileViewModel()` to get `ProfileViewModel` by using a `ProfileViewModelFactory`):
+
+```kotlin
+
+class ProfileViewModel(
+    private val getProfileLikeCountUseCase: GetProfileLikeCountUseCase,
+    private val navArgs: ProfileScreenNavArgs
+): ViewModel(){
+    ...
+}
+
+class ProfileViewModelFactory(
+    private val getProfileLikeCountUseCase: GetProfileLikeCountUseCase
+): AbstractSavedStateViewModelFactory() {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel?> create(
+        key: String,
+        modelClass: Class<T>,
+        handle: SavedStateHandle
+    ): T {
+        return ProfileViewModel(
+            getProfileLikeCountUseCase,
+            ProfileScreenDestination.argsFrom(handle)
+        ) as T
+    }
+}
+
+@Composable
+fun profileViewModel(): ProfileViewModel {
+    val dependencyContainer = [ACCESS YOUR DEPENDENCIES GRAPH HERE SOMEHOW]
+    val factory = ProfileViewModelFactory(dependencyContainer.getProfileLikeCountUseCase)
+    return viewModel(factory = factory)
```

Comment:
I believe you're missing here to also pass the default args, this is used when you want to then use SavedStateHandle to get navigation arguments. Same as my example above this one.

`defaultArgs = (savedStateRegistryOwner as? NavBackStackEntry)?.arguments` pass it to the Factory, and them pass it to `AbstractSavedStateViewModelFactory` constructor.

Same thing for the `SavedStateRegistryOwner`, I believe you need to use it in `AbstractSavedStateViewModelFactory` explicitly so that the owner is respected.